client/tests/integration/time_trigger.rs

```diff
@@ -0,0 +1,116 @@
+#![allow(clippy::restriction)]
+
+use std::time::Duration;
+
+use eyre::Result;
+use iroha_client::client::{self, Client};
+use iroha_core::block::DEFAULT_CONSENSUS_ESTIMATION_MS;
+use iroha_data_model::prelude::*;
+use test_network::{Peer as TestPeer, *};
+
+/// Macro to abort compilation, if `e` isn't `true`
+macro_rules! const_assert {
+    ($e:expr) => {
+        #[allow(trivial_casts)]
+        const _: usize = ($e as bool) as usize - 1;
+    };
+}
+
+/// Time-based triggers and block commitment process depend heavily on **current time** and **CPU**,
+/// so it's impossible to create fully reproducible test scenario.
+/// But it's possible to compare expectations an reality about trigger execution count. In this test
+/// it is expected to have estimation error < **10%**
+#[test]
+#[allow(clippy::cast_precision_loss)]
+fn time_trigger_execution_count_error_should_be_less_than_10_percent() -> Result<()> {
+    const PERIOD_MS: u64 = 100;
+    const ACCEPTABLE_ERROR_PERCENT: u8 = 10;
+    const_assert!(PERIOD_MS < DEFAULT_CONSENSUS_ESTIMATION_MS);
+    const_assert!(ACCEPTABLE_ERROR_PERCENT <= 100);
+
+    let (_rt, _peer, mut test_client) = <TestPeer>::start_test_with_runtime();
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+    let start_time = current_time();
+
+    let asset_definition_id = AssetDefinitionId::new("rose", "wonderland").expect("Valid");
+    let account_id = AccountId::new("alice", "wonderland").expect("Valid");
+    let asset_id = AssetId::new(asset_definition_id, account_id.clone());
+
+    let prev_value = test_client
+        .request(client::asset::by_id(asset_id.clone()))?
+        .value;
+
+    let schedule =
+        TimeSchedule::starting_at(start_time).with_period(Duration::from_millis(PERIOD_MS));
+    let register_trigger =
+        build_register_trigger_isi(account_id.clone(), asset_id.clone(), schedule)?;
+    test_client.submit(register_trigger)?;
+
+    submit_sample_isi_on_every_block_commit(&mut test_client, &account_id, 3)?;
+    std::thread::sleep(Duration::from_millis(DEFAULT_CONSENSUS_ESTIMATION_MS));
+
+    let finish_time = current_time();
+    let average_count = finish_time.saturating_sub(start_time).as_millis() / u128::from(PERIOD_MS);
+
+    let output_asset = test_client.request(client::asset::by_id(asset_id))?;
+    let actual_value = *TryAsRef::<u32>::try_as_ref(&output_asset)?;
+    let expected_value = TryAsRef::<u32>::try_as_ref(&prev_value)? + u32::try_from(average_count)?;
+    let acceptable_error = expected_value as f32 * (f32::from(ACCEPTABLE_ERROR_PERCENT) / 100.0);
+    let error = i32::abs((actual_value - expected_value).try_into()?) as f32;
```

Comment:
Can underflow in usual cases
```suggestion
    let error = (core::cmp::max(actual_value, expected_value) - core::cmp::min(actual_value, expected_value)) as f32;
```